resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala

```diff
@@ -433,4 +442,39 @@ class ClientSuite extends SparkFunSuite with Matchers {
     classpath(env)
   }
 
+  private def testResourceRequest(expectedResources: Seq[(String, Long)],
```

Comment:
Repeating the comment since github doesn't re-open the original thread:

This isn't fixed yet. You're still passing in the expected resources, which are expected to match the hardcoded configs that are set in this method

Reply:
Fair enough. I decided to pass in the sparkConf, the list of known resources and the expected resources and their values, so the testcase is more clear now.